packages/react-components/src/organisms/ToolModal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import { TeamTool } from '@asap-hub/model';
+
+import ModalEditHeader from '../molecules/ModalEditHeader';
+import { Modal, LabeledTextField } from '../molecules';
+import { noop } from '../utils';
+
+type ToolModalProps = Partial<TeamTool> & {
+  onSave?: (data: TeamTool) => void;
+  title: string;
+  backHref: string;
+};
+
+const ToolModal: React.FC<ToolModalProps> = ({
+  title,
+  url = '',
+  description = '',
+  name = '',
+  onSave = noop,
+  backHref,
+}) => {
+  const [newUrl, setNewUrl] = useState(url);
+  const [newDescription, setNewDescription] = useState(description);
+  const [newName, setNewName] = useState(name);
+  return (
+    <Modal>
+      <ModalEditHeader
```

Comment:
There should be a form around the header and fields. The save button is automatically already `type="submit"`

Reply:
search `reportValidity` for form validation in save handlers